lambdas/bulk-migrate-vcs/src/main/java/uk/gov/di/ipv/core/bulkmigratevcs/BulkMigrateVcsHandler.java

```diff
@@ -282,15 +330,35 @@ private void migrateIdentity(
                     e,
                     batchId,
                     pageSummary);
-            // Send failed audit event
+            auditService.sendAuditEvent(
+                    AuditEvent.createWithoutDeviceInformation(
+                            AuditEventTypes.IPV_EVCS_MIGRATION_FAILURE,
+                            configService.getParameter(ConfigurationVariable.COMPONENT_ID),
+                            new AuditEventUser(reportItem.getHashUserId(), "", "", ""),
+                            new AuditExtensionsEvcsFailedMigration(
+                                    batchId, vcs.size(), e.getMessage())));
             pageSummary.incrementFailedEvcsWrite(reportItem.getHashUserId());
             return;
         }
 
         try {
             setMigratedOnTacticalStoreVcs(vcs, timestamp);
             LOGGER.info(annotateLog("Migrated", reportItem, batchId, pageSummary));
-            // Send migrated audit event
+            List<String> vcsCredentials =
+                    vcs.stream()
+                            .map(
+                                    vc -> {
+                                        String[] parts = vc.getVcString().split("\\.");
+                                        return parts.length > 2 ? parts[2] : "";
+                                    })
+                            .toList();
```

Comment:
I thinkj this could be streamlined a bit - we should be pretty sure our vc strings are in good shape.

```suggestion
            var vcsCredentials = vcs.stream().map(vc -> vc.getVcString().split("\\.")[2]).toList();
```

You could even inline it into the method below.

Reply:
I used single line one initially, but one of the test was failing with `java.lang.ArrayIndexOutOfBoundsException` may be I need to update data in the test setup.